### PR TITLE
dev-python/tablib: add package for www-apps/netbox-2.11.0

### PR DESCRIPTION
**Pull request by scardracs**
List of new packages:
dev-python/tablib for www-apps/netbox-2.11.0+
dev-python/markuppy for testing pourpouse on tablib (required only for tests)

@anthonyryan1 tell me if this helps for netbox

Package-Manager: Portage-3.0.18, Repoman-3.0.3
Signed-off-by: Marco Scardovi <marco@scardovi.com>

### CODE REVIEW

**Review by mgorny — CHANGES_REQUESTED** [review]
Tests, tests, tests.

File: dev-python/tablib/tablib-3.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+SLOT="0"
+LICENSE="BSD-2"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="test"
+
+BDEPEND="
+	test? (
+		dev-python/pytest
+		dev-python/pytest-cov
```
Comment (mgorny) [inline dev-python/tablib/tablib-3.0.0.ebuild:22]: We don't add that. Strip the dep if it is really required.

File: dev-python/tablib/tablib-3.0.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		dev-python/odfpy
+		dev-python/openpyxl
+		dev-python/markuppy
+		dev-python/pandas
+		dev-python/pyyaml
+		dev-python/tabulate
+		dev-python/xlrd
+		dev-python/xlwt
```
Comment (mgorny) [inline dev-python/tablib/tablib-3.0.0.ebuild:30]: Where are all the PYTHON_USEDEPs?

File: dev-python/tablib/tablib-3.0.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+SLOT="0"
+LICENSE="BSD-2"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="test"
+
+BDEPEND="
+	test? (
+		dev-python/pytest
```
Comment (mgorny) [inline dev-python/tablib/tablib-3.0.0.ebuild:21]: Explicit dep on pytest is redundant if you use det.

File: dev-python/tablib/tablib-3.0.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+		dev-python/tabulate
+		dev-python/xlrd
+		dev-python/xlwt
+	)
+"
+
+distutils_enable_tests pytest
+
+python_test() {
+	epytest --deselect MarkupPy
```
Comment (mgorny) [inline dev-python/tablib/tablib-3.0.0.ebuild:37]: Why do you deselect it if you added it as a dep anyway?

File: dev-python/tablib/tablib-3.0.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+inherit distutils-r1
+
+DESCRIPTION="Format-agnostic tabular dataset library"
+HOMEPAGE="https://pypi.org/project/tablib/ https://github.com/jazzband/tablib/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+SLOT="0"
+LICENSE="BSD-2"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
```
Comment (mgorny) [inline dev-python/tablib/tablib-3.0.0.ebuild:16]: Did you actually test it on Prefix? I'm not saying adding prefix keywords is entirely wrong but it is weird when almost no deps have them.

File: dev-python/tablib/tablib-3.0.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+IUSE="test"
```
Comment (mgorny) [inline dev-python/tablib/tablib-3.0.0.ebuild:17]: Redundant.

File: dev-python/markuppy/markuppy-1.1.4.ebuild [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7..9} )
+
+inherit distutils-r1
+
+DESCRIPTION="Format-agnostic tabular dataset library"
+HOMEPAGE="https://pypi.org/project/MarkupPy/ https://github.com/tylerbakke/MarkupPy"
+SRC_URI="https://files.pythonhosted.org/packages/4e/ca/f43541b41bd17fc945cfae7ea44f1661dc21ea65ecc944a6fa138eead94c/MarkupPy-1.14.tar.gz -> ${P}.tar.gz"
```
Comment (mgorny) [inline dev-python/markuppy/markuppy-1.1.4.ebuild:12]: We have `mirror://pypi` for that.

File: dev-python/markuppy/markuppy-1.1.4.ebuild [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+SLOT="0"
+LICENSE="BSD-2"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+
+# MarkupPy does not have any test suite
+RESTRICT="test"
```
Comment (mgorny) [inline dev-python/markuppy/markuppy-1.1.4.ebuild:19]: Are you sure packaging it is a good idea? Given that upstream didn't manage to actually push code into the repository, and didn't do anything for 3.5 years about it, I feel like packaging it is a waste of time.

File: dev-python/tablib/tablib-3.0.0.ebuild (hunk 4 above)
Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild:37]: Sorry, typo

File: dev-python/tablib/tablib-3.0.0.ebuild (hunk 2 above)
Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild:30]: Added

File: dev-python/tablib/tablib-3.0.0.ebuild (hunk 4 above)
Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild:37]: Sorry, typo

File: dev-python/tablib/tablib-3.0.0.ebuild (hunk 5 above)
Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild:16]: Changed with only ~amd64

File: dev-python/tablib/tablib-3.0.0.ebuild (hunk 3 above)
Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild:21]: Removed

File: dev-python/tablib/tablib-3.0.0.ebuild (hunk 6 above)
Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild:17]: Fixed

File: dev-python/tablib/tablib-3.0.0.ebuild (hunk 1 above)
Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild:22]: Do you mean pytest and pytest-cov?

Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild:22]: Nvm

File: dev-python/markuppy/markuppy-1.1.4.ebuild (hunk 8 above)
Comment (scardracs) [reply dev-python/markuppy/markuppy-1.1.4.ebuild:19]: tablib is required for newer versions of netbox, this is why I'm packaging it

**Comment by anthonyryan1** [comment]
The tablib ebuild looks good to me.

**Comment by scardracs** [comment]
@mgorny I was waiting for fixed gentoo-repo-qa-bot before pushing fixes

File: dev-python/tablib/tablib-3.0.0.ebuild (hunk 1 above)
Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild]: dev-python/pytest-cov is required so I leave it as is

File: dev-python/tablib/tablib-3.0.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+HOMEPAGE="https://pypi.org/project/tablib/ https://github.com/jazzband/tablib/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+SLOT="0"
+LICENSE="BSD-2"
+KEYWORDS="~amd64"
+
+BDEPEND="
+	test? (
+		dev-python/pytest-cov[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/tablib/tablib-3.0.0.ebuild:20]: What is it required for, then?

Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild:20]: Yes. Removing this will fails on the way you reported yesterday

Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild:20]: Ah sorry, I have read wait instead of what 😅

Comment (mgorny) [reply dev-python/tablib/tablib-3.0.0.ebuild:20]: Did you sed out the unnecessary pytest args as described in the docs and found in a few dozen other ebuilds?

Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild:20]: Ok i have seen what you mean, i will fix it asap

Comment (scardracs) [reply dev-python/tablib/tablib-3.0.0.ebuild]: @mgorny fixed

**Comment by scardracs** [comment]
@mgorny what about unstaged commits?

**Pull request closed without merging**